### PR TITLE
feat: add check for hardened_malloc flatpak preload

### LINKED ISSUE #411: [FEAT] audit-secureblue should check flatpaks for hardened_malloc preload

### PR DESCRIPTION
**Pull request by Rubiginosa**
Chose failure for no preload based on design principles—incompatibility with hardened_malloc is more a flaw in an application than it is a necessity like network access.

Also added some ternary logic for `status` to make things a little more extensible for future checks.

Closes #411

### CODE REVIEW

File: files/system/usr/share/ublue-os/just/70-secureblue.just [hunk 1]
```diff
@@ -534,28 +534,34 @@ audit-secureblue:
...
+                status="$STATUS_FAILURE"
             fi
-            flatpak_test_string="Auditing $f"
-            if [[ ! $has_network == "true" && ! $has_x11 == "true" ]]; then
-                print_status "$flatpak_test_string" "$STATUS_SUCCESS"
-            elif [[ $has_x11 == "true" ]]; then
-                print_status "$flatpak_test_string" "$STATUS_FAILURE"
-            elif [[ $has_network == "true" ]]; then
-                print_status "$flatpak_test_string" "$STATUS_WARNING"
+            if ! hasPermission "$permissions" "LD_PRELOAD" "/var/run/host/usr/lib64/libhardened_malloc.so"; then
```
Comment (RoyalOughtness) [inline files/system/usr/share/ublue-os/just/70-secureblue.just]: this won't work because by default we use the optimized library

File: files/system/usr/share/ublue-os/just/70-secureblue.just [hunk 2]
```diff
@@ -534,28 +534,34 @@ audit-secureblue:
...
                 has_network=true
+                [[ ! "$status" != "$STATUS_FAILURE" ]] && status="$STATUS_WARNING"
             fi
-            if hasPermission "$permissions" "sockets" "x11" && ! hasPermission "$permissions" "sockets" "fallback-x11"  ]]; then
+            if hasPermission "$permissions" "sockets" "x11" && ! hasPermission "$permissions" "sockets" "fallback-x11"; then
                 has_x11=true
+                status="$STATUS_FAILURE"
             fi
-            flatpak_test_string="Auditing $f"
-            if [[ ! $has_network == "true" && ! $has_x11 == "true" ]]; then
```
Comment (RoyalOughtness) [inline files/system/usr/share/ublue-os/just/70-secureblue.just]: why was this removed?

File: files/system/usr/share/ublue-os/just/70-secureblue.just [hunk 3]
```diff
@@ -534,28 +534,34 @@ audit-secureblue:
         for f in ${!flatpaks[@]}; do
             has_network=false
             has_x11=false
+            has_hardened_malloc=true
+            status=$STATUS_SUCCESS
             fullref=${flatpaks["$f"]}
             permissions=$(flatpak info --show-permissions "$fullref")
             if hasPermission "$permissions" "shared" "network"; then
                 has_network=true
+                [[ ! "$status" != "$STATUS_FAILURE" ]] && status="$STATUS_WARNING"
```
Comment (RoyalOughtness) [inline files/system/usr/share/ublue-os/just/70-secureblue.just]: what is this doing? double negative :smile:

File: files/system/usr/share/ublue-os/just/70-secureblue.just [hunk 4]
```diff
@@ -534,28 +534,34 @@ audit-secureblue:
...
             fi
-            if hasPermission "$permissions" "sockets" "x11" && ! hasPermission "$permissions" "sockets" "fallback-x11"  ]]; then
+            if hasPermission "$permissions" "sockets" "x11" && ! hasPermission "$permissions" "sockets" "fallback-x11"; then
```
Comment (RoyalOughtness) [inline files/system/usr/share/ublue-os/just/70-secureblue.just]: can you put the changes that aren't related to hardened_malloc flatpak checks in another PR to make this easier to review?

File: files/system/usr/share/ublue-os/just/70-secureblue.just (hunk 1 above)
Comment (Rubiginosa) [reply files/system/usr/share/ublue-os/just/70-secureblue.just]: I see. Should the script check specifically for matching architecture, or just any hardened_malloc?

File: files/system/usr/share/ublue-os/just/70-secureblue.just (hunk 2 above)
Comment (Rubiginosa) [reply files/system/usr/share/ublue-os/just/70-secureblue.just]: Part of the change to how the status is managed. Rather than checking something like 2^n states (less in reality, but still), each permission check can only move the status in one direction: SUCCESS > WARNING or FAILURE, but never FAILURE > WARNING or SUCCESS.

File: files/system/usr/share/ublue-os/just/70-secureblue.just (hunk 4 above)
Comment (Rubiginosa) [reply files/system/usr/share/ublue-os/just/70-secureblue.just]: Yep, can do!

File: files/system/usr/share/ublue-os/just/70-secureblue.just (hunk 1 above)
Comment (RoyalOughtness) [reply files/system/usr/share/ublue-os/just/70-secureblue.just]: just the string "libhardened_malloc" is probably sufficient

File: files/system/usr/share/ublue-os/just/70-secureblue.just (hunk 2 above)
Comment (RoyalOughtness) [reply files/system/usr/share/ublue-os/just/70-secureblue.just]: interesting, thanks for clarifying

**Review by RoyalOughtness — APPROVED** [review]
